compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1960,12 +1965,11 @@ public void visit(BLangSimpleVarRef bLangSimpleVarRef) {
                 bLangSimpleVarRef.symbol = symbol;
                 bLangSimpleVarRef.varSymbol = symbol;
             }
-        } else if (resolvedSymbol != symTable.notFoundSymbol && symbol != null) {
+        } else if (resolvedSymbol != symTable.notFoundSymbol) {
             resolvedSymbol.closure = true;
             // When there's a type guard, there can be a enclSymbol before type narrowing.
             // So, we have to mark that as a closure as well.
-            BSymbol enclSymbol = symResolver.lookupClosureVarSymbol(env.enclEnv,
-                    Names.fromString(identifier), SymTag.VARIABLE);
+            BSymbol enclSymbol = symResolver.lookupClosureVarSymbol(env.enclEnv, symbol);
             if (enclSymbol != null && enclSymbol != symTable.notFoundSymbol) {
                 enclSymbol.closure = true;
             }
```

Comment:
Since we obtain the resolved symbol directly from the original symbol, these checks are no longer necessary. We can remove them.